### DISCUSSION TITLE
Need for ptrimmer

### DISCUSSION THREAD

**Discussion by gregdenay** [bio-raum/FooDMe2#17] [General]
@marchoeppner 
Hi, cutadapt seems to support reverse cmplementation, IUPAC ambiguous bases and multiple trimming in a single step + MultiQC ready reports. I need to test it but if it works as I think it does I could simplify the trimming subworkflow.
Do we need to keep ptrimmer around (For ONT or IonTorrent?) or is it completely redundant?

**Comment by marchoeppner** [comment]
I think it is basically redundant at this point. We could think about how we can remove the need for the `--cutadapt_trim_3p` option though - seems like an unnecessary source of errors. That was the main reason why I used Ptrimmer originally, since it knows about the expected amplicon size and would trim both ends if need be. 

Maybe read out the max length of the reads and compare that to a pre-configured amplicon size to determine whether trimming should be done on both ends.  But that would make it more difficult for users to use their own primers..

**Reply by gregdenay** [reply]
Maybe we can use the insert peak size (+ some tolerance) from fastp instead of predefining the expected size?

**Reply by marchoeppner** [reply]
Right, we would still need to know the size of the amplicon though, which we can easily add to conf/primers.conf. but if one wanted to use a custom set of primers , then we need an additional command line option `--amplicon_size` or similar. Not sure that is a great improvement ;D
